marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/default/main.sjs

```diff
@@ -87,6 +87,10 @@ function main(content, options) {
 
   //now let's make our attachments, if it's xml, it'll be passed as string
   instance['$attachments'] = doc;
+  // fix the document URI if the format changes
+  if (content.uri && !content.uri.endsWith(outputFormat)) {
```

Comment:
Is outputFormat known to be lowercase here? I don't know if it matters if "XML" or "JSON" is used?